src/core/manager.c

```diff
@@ -3963,6 +3959,11 @@ void manager_check_finished(Manager *m) {
                 return;
         }
 
+        /* The jobs hashmap tends to grow a lot during boot, and then it's not reused until shutdown. Let's
+           kill the hashmap if it is relatively large. */
+        if (hashmap_buckets(m->jobs) > hashmap_size(m->units) / 10)
+                m->jobs = hashmap_free(m->jobs);
```

Comment:
hmmm, a few hundred buckets too many shouldn't matter should they?

I can see why adding this to the hashmap implementation makes sense, because then this pays off on every hashmap. but for this single one? sounds like a lot of fuss for very little effect. but ok